Src/DependencyCollector/Shared.Tests/Implementation/Operation/CacheBasedOperationHolderTests.cs

```diff
@@ -95,6 +96,19 @@ public void GetReturnsItemIfItExistsInTheTable()
             Assert.AreEqual(this.telemetryTuple, this.cacheBasedOperationHolder.Get(id));
         }
 
+        [TestMethod]
+        public void TestItemExpiration()
+        {
+            long id = 911911;
+            this.cacheBasedOperationHolder.Store(id, this.telemetryTuple);
+            Assert.AreEqual(this.telemetryTuple, this.cacheBasedOperationHolder.Get(id));
+
+            // Sleep for 2 secs which is more than the cache expiration time.
+            Thread.Sleep(2000);
```

Comment:
Agreed. Will change. 
Regarding timing, i don't think i can go below 1sec as then it might cause random failures since the exact sleep time is not guaranteed.